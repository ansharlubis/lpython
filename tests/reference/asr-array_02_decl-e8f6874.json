{
    "basename": "asr-array_02_decl-e8f6874",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/array_02_decl.py",
    "infile_hash": "49bfd0530662b64164eb87c7e1bc53d518a7ccc75469dfb7ced80e5f",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-array_02_decl-e8f6874.stdout",
    "stdout_hash": "117b8e9eaf90cd1f3b95204f45cd431d8010d0cb33e369dfb644ffe7",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}