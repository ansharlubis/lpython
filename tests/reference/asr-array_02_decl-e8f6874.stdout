(TranslationUnit (SymbolTable 1 {_lpython_main_program: (Subroutine (SymbolTable 18 {}) _lpython_main_program [] [(SubroutineCall 1 declare_arrays () [] ())] Source Public Implementation () .false. .false.), accept_multidim_f32_array: (Function (SymbolTable 14 {_lpython_return_variable: (Variable 14 _lpython_return_variable ReturnVar () () Default (Real 4 []) Source Public Required .false.), xf32: (Variable 14 xf32 InOut () () Default (Real 4 [(() ())]) Source Public Required .false.)}) accept_multidim_f32_array [(Var 14 xf32)] [(= (Var 14 _lpython_return_variable) (ArrayItem (Var 14 xf32) [(() (IntegerConstant 0 (Integer 4 [])) ())] (Real 4 []) ()) ()) (Return)] (Var 14 _lpython_return_variable) Source Public Implementation .false. ()), accept_multidim_f64_array: (Function (SymbolTable 15 {_lpython_return_variable: (Variable 15 _lpython_return_variable ReturnVar () () Default (Real 8 []) Source Public Required .false.), xf64: (Variable 15 xf64 InOut () () Default (Real 8 [(() ()) (() ())]) Source Public Required .false.)}) accept_multidim_f64_array [(Var 15 xf64)] [(= (Var 15 _lpython_return_variable) (ArrayItem (Var 15 xf64) [(() (IntegerConstant 0 (Integer 4 [])) ()) (() (IntegerConstant 1 (Integer 4 [])) ())] (Real 8 []) ()) ()) (Return)] (Var 15 _lpython_return_variable) Source Public Implementation .false. ()), accept_multidim_i32_array: (Function (SymbolTable 12 {_lpython_return_variable: (Variable 12 _lpython_return_variable ReturnVar () () Default (Integer 4 []) Source Public Required .false.), xi32: (Variable 12 xi32 InOut () () Default (Integer 4 [(() ()) (() ())]) Source Public Required .false.)}) accept_multidim_i32_array [(Var 12 xi32)] [(= (Var 12 _lpython_return_variable) (ArrayItem (Var 12 xi32) [(() (IntegerConstant 0 (Integer 4 [])) ()) (() (IntegerConstant 0 (Integer 4 [])) ())] (Integer 4 []) ()) ()) (Return)] (Var 12 _lpython_return_variable) Source Public Implementation .false. ()), accept_multidim_i64_array: (Function (SymbolTable 13 {_lpython_return_variable: (Variable 13 _lpython_return_variable ReturnVar () () Default (Integer 8 []) Source Public Required .false.), xi64: (Variable 13 xi64 InOut () () Default (Integer 8 [(() ()) (() ()) (() ())]) Source Public Required .false.)}) accept_multidim_i64_array [(Var 13 xi64)] [(= (Var 13 _lpython_return_variable) (ArrayItem (Var 13 xi64) [(() (IntegerConstant 9 (Integer 4 [])) ()) (() (IntegerConstant 9 (Integer 4 [])) ()) (() (IntegerConstant 9 (Integer 4 [])) ())] (Integer 8 []) ()) ()) (Return)] (Var 13 _lpython_return_variable) Source Public Implementation .false. ()), declare_arrays: (Subroutine (SymbolTable 16 {ac32: (Variable 16 ac32 Local () () Default (Complex 4 [((IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (IntegerBinOp (IntegerBinOp (IntegerConstant 3 (Integer 4 [])) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 2 (Integer 4 []))) Sub (IntegerConstant 0 (Integer 4 [])) (Integer 4 []) ()) Add (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 3 (Integer 4 [])))) ((IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (IntegerBinOp (IntegerBinOp (IntegerConstant 5 (Integer 4 [])) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 4 (Integer 4 []))) Sub (IntegerConstant 0 (Integer 4 [])) (Integer 4 []) ()) Add (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 5 (Integer 4 [])))) ((IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (IntegerBinOp (IntegerBinOp (IntegerConstant 99 (Integer 4 [])) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 98 (Integer 4 []))) Sub (IntegerConstant 0 (Integer 4 [])) (Integer 4 []) ()) Add (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 99 (Integer 4 []))))]) Source Public Required .false.), ac64: (Variable 16 ac64 Local () () Default (Complex 8 [((IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (IntegerBinOp (IntegerBinOp (IntegerConstant 10 (Integer 4 [])) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 9 (Integer 4 []))) Sub (IntegerConstant 0 (Integer 4 [])) (Integer 4 []) ()) Add (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 10 (Integer 4 [])))) ((IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (IntegerBinOp (IntegerBinOp (IntegerConstant 13 (Integer 4 [])) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 12 (Integer 4 []))) Sub (IntegerConstant 0 (Integer 4 [])) (Integer 4 []) ()) Add (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 13 (Integer 4 [])))) ((IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (IntegerBinOp (IntegerBinOp (IntegerConstant 11 (Integer 4 [])) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 10 (Integer 4 []))) Sub (IntegerConstant 0 (Integer 4 [])) (Integer 4 []) ()) Add (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 11 (Integer 4 [])))) ((IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (IntegerBinOp (IntegerBinOp (IntegerConstant 16 (Integer 4 [])) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 15 (Integer 4 []))) Sub (IntegerConstant 0 (Integer 4 [])) (Integer 4 []) ()) Add (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 16 (Integer 4 []))))]) Source Public Required .false.), af32: (Variable 16 af32 Local () () Default (Real 4 [((IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (IntegerBinOp (IntegerBinOp (IntegerConstant 3 (Integer 4 [])) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 2 (Integer 4 []))) Sub (IntegerConstant 0 (Integer 4 [])) (Integer 4 []) ()) Add (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 3 (Integer 4 []))))]) Source Public Required .false.), af64: (Variable 16 af64 Local () () Default (Real 8 [((IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (IntegerBinOp (IntegerBinOp (IntegerConstant 10 (Integer 4 [])) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 9 (Integer 4 []))) Sub (IntegerConstant 0 (Integer 4 [])) (Integer 4 []) ()) Add (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 10 (Integer 4 [])))) ((IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (IntegerBinOp (IntegerBinOp (IntegerConstant 4 (Integer 4 [])) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 3 (Integer 4 []))) Sub (IntegerConstant 0 (Integer 4 [])) (Integer 4 []) ()) Add (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 4 (Integer 4 []))))]) Source Public Required .false.), ai32: (Variable 16 ai32 Local () () Default (Integer 4 [((IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (IntegerBinOp (IntegerBinOp (IntegerConstant 3 (Integer 4 [])) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 2 (Integer 4 []))) Sub (IntegerConstant 0 (Integer 4 [])) (Integer 4 []) ()) Add (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 3 (Integer 4 [])))) ((IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (IntegerBinOp (IntegerBinOp (IntegerConstant 3 (Integer 4 [])) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 2 (Integer 4 []))) Sub (IntegerConstant 0 (Integer 4 [])) (Integer 4 []) ()) Add (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 3 (Integer 4 []))))]) Source Public Required .false.), ai64: (Variable 16 ai64 Local () () Default (Integer 8 [((IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (IntegerBinOp (IntegerBinOp (IntegerConstant 10 (Integer 4 [])) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 9 (Integer 4 []))) Sub (IntegerConstant 0 (Integer 4 [])) (Integer 4 []) ()) Add (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 10 (Integer 4 [])))) ((IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (IntegerBinOp (IntegerBinOp (IntegerConstant 10 (Integer 4 [])) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 9 (Integer 4 []))) Sub (IntegerConstant 0 (Integer 4 [])) (Integer 4 []) ()) Add (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 10 (Integer 4 [])))) ((IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (IntegerBinOp (IntegerBinOp (IntegerConstant 10 (Integer 4 [])) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 9 (Integer 4 []))) Sub (IntegerConstant 0 (Integer 4 [])) (Integer 4 []) ()) Add (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 10 (Integer 4 []))))]) Source Public Required .false.)}) declare_arrays [] [(Print () [(FunctionCall 1 accept_multidim_i32_array () [((Var 16 ai32))] (Integer 4 []) () ())] () ()) (Print () [(FunctionCall 1 accept_multidim_i64_array () [((Var 16 ai64))] (Integer 8 []) () ())] () ()) (Print () [(FunctionCall 1 accept_multidim_f32_array () [((Var 16 af32))] (Real 4 []) () ())] () ()) (Print () [(FunctionCall 1 accept_multidim_f64_array () [((Var 16 af64))] (Real 8 []) () ())] () ())] Source Public Implementation () .false. .false.), main_program: (Program (SymbolTable 17 {}) main_program [] [(SubroutineCall 1 _lpython_main_program () [] ())]), numpy: (Module (SymbolTable 3 {__lpython_overloaded_0__cos: (Function (SymbolTable 9 {_lpython_return_variable: (Variable 9 _lpython_return_variable ReturnVar () () Default (Real 8 []) Source Public Required .false.), x: (Variable 9 x In () () Default (Real 8 []) Source Public Required .false.)}) __lpython_overloaded_0__cos [(Var 9 x)] [(= (Var 9 _lpython_return_variable) (FunctionCall 3 _lfortran_dcos () [((Var 9 x))] (Real 8 []) () ()) ()) (Return)] (Var 9 _lpython_return_variable) Source Public Implementation .true. ()), __lpython_overloaded_0__sin: (Function (SymbolTable 5 {_lpython_return_variable: (Variable 5 _lpython_return_variable ReturnVar () () Default (Real 8 []) Source Public Required .false.), x: (Variable 5 x In () () Default (Real 8 []) Source Public Required .false.)}) __lpython_overloaded_0__sin [(Var 5 x)] [(= (Var 5 _lpython_return_variable) (FunctionCall 3 _lfortran_dsin () [((Var 5 x))] (Real 8 []) () ()) ()) (Return)] (Var 5 _lpython_return_variable) Source Public Implementation .true. ()), __lpython_overloaded_1__cos: (Function (SymbolTable 11 {_lpython_return_variable: (Variable 11 _lpython_return_variable ReturnVar () () Default (Real 4 []) Source Public Required .false.), x: (Variable 11 x In () () Default (Real 4 []) Source Public Required .false.)}) __lpython_overloaded_1__cos [(Var 11 x)] [(= (Var 11 _lpython_return_variable) (FunctionCall 3 _lfortran_scos () [((Var 11 x))] (Real 4 []) () ()) ()) (Return)] (Var 11 _lpython_return_variable) Source Public Implementation .true. ()), __lpython_overloaded_1__sin: (Function (SymbolTable 7 {_lpython_return_variable: (Variable 7 _lpython_return_variable ReturnVar () () Default (Real 4 []) Source Public Required .false.), x: (Variable 7 x In () () Default (Real 4 []) Source Public Required .false.)}) __lpython_overloaded_1__sin [(Var 7 x)] [(= (Var 7 _lpython_return_variable) (FunctionCall 3 _lfortran_ssin () [((Var 7 x))] (Real 4 []) () ()) ()) (Return)] (Var 7 _lpython_return_variable) Source Public Implementation .true. ()), _lfortran_dcos: (Function (SymbolTable 8 {_lpython_return_variable: (Variable 8 _lpython_return_variable ReturnVar () () Default (Real 8 []) BindC Public Required .false.), x: (Variable 8 x In () () Default (Real 8 []) BindC Public Required .true.)}) _lfortran_dcos [(Var 8 x)] [] (Var 8 _lpython_return_variable) BindC Public Interface .false. ()), _lfortran_dsin: (Function (SymbolTable 4 {_lpython_return_variable: (Variable 4 _lpython_return_variable ReturnVar () () Default (Real 8 []) BindC Public Required .false.), x: (Variable 4 x In () () Default (Real 8 []) BindC Public Required .true.)}) _lfortran_dsin [(Var 4 x)] [] (Var 4 _lpython_return_variable) BindC Public Interface .false. ()), _lfortran_scos: (Function (SymbolTable 10 {_lpython_return_variable: (Variable 10 _lpython_return_variable ReturnVar () () Default (Real 4 []) BindC Public Required .false.), x: (Variable 10 x In () () Default (Real 4 []) BindC Public Required .true.)}) _lfortran_scos [(Var 10 x)] [] (Var 10 _lpython_return_variable) BindC Public Interface .false. ()), _lfortran_ssin: (Function (SymbolTable 6 {_lpython_return_variable: (Variable 6 _lpython_return_variable ReturnVar () () Default (Real 4 []) BindC Public Required .false.), x: (Variable 6 x In () () Default (Real 4 []) BindC Public Required .true.)}) _lfortran_ssin [(Var 6 x)] [] (Var 6 _lpython_return_variable) BindC Public Interface .false. ()), cos: (GenericProcedure 3 cos [3 __lpython_overloaded_0__cos 3 __lpython_overloaded_1__cos] Public), sin: (GenericProcedure 3 sin [3 __lpython_overloaded_0__sin 3 __lpython_overloaded_1__sin] Public)}) numpy [] .false. .false.)}) [])
